cfn_lsp_extra/cfnlint_integration.py

```diff
@@ -32,6 +32,21 @@ def diagnostics(yaml_content: str, file_path: str) -> List[Diagnostic]:
     regions = ["us-east-1"]
 
     if not errors:
+        try:
+            (config, _, _) = cfnlint.core.get_args_filenames(file_path)
+            # override rules with config-specific rules
+            rules = cfnlint.core.get_rules(
+                config.append_rules,
+                config.ignore_checks,
+                config.include_checks,
+                config.configure_rules,
+                config.include_experimental,
+                config.mandatory_checks,
+                config.custom_rules,
+            )
+        except Exception:
```

Comment:
I think this maybe should be `BaseException` (see https://docs.python.org/3/library/exceptions.html#SystemExit) since `get_args_filenames` [appears](https://github.com/aws-cloudformation/cfn-lint/blob/1d5c4de28c18a6d2a4d5d72f48d28434f1b0bc8e/src/cfnlint/core.py#L206) to call `sys.exit` if the config file is invalid.

Or perhaps `get_args_filenames` could be switched out with `ConfigMixIn`.

Reply:
Ahh I'm following cfn-lint a bit better, now it's just getting the `ConfigMixIn` directly, and it still seems to work on my end. From what I could tell there's one area where a `sys.exit` call is made and that's just in their `ArgumentParser` method override, which I don't think we'd run into? Happy to switch it to whatever you feel is best though